src/NLog/LayoutRenderers/GdcLayoutRenderer.cs

```diff
@@ -57,8 +57,8 @@ public class GdcLayoutRenderer : LayoutRenderer
         /// <param name="logEvent">Logging event.</param>
         protected override void Append(StringBuilder builder, LogEventInfo logEvent)
         {
-            string msg = GlobalDiagnosticsContext.Get(this.Item);
-            builder.Append(msg);
+            // use AppendFormat to ensure that any IFormatProvider is respected
+            builder.AppendFormat(logEvent.FormatProvider, "{0}", GlobalDiagnosticsContext.GetObject(this.Item));
```

Comment:
This looks a bit strange. (string.format and "{0}")

I propose: `GlobalDiagnosticsContext.Get(this.item, logEvent.FormatProvider). - (so a new overload)